libs/platform/ebpf_serialize.h

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+// Data structures and functions for serializing and de-serializing eBPF structs
+// between API and Execution Context.
+
+#pragma once
+
+#include "ebpf_platform.h"
+#include "ebpf_core_structs.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+    /**
+     * @brief eBPF Core Map Information
+     */
+    typedef struct _ebpf_core_map_information
+    {
+        ebpf_map_definition_t definition;
+        ebpf_utf8_string_t pin_path;
+    } ebpf_core_map_information_t;
+
+    /**
+     * @brief Serialized eBPF Map Information.
+     */
+    typedef struct _ebpf_serialized_map_information
+    {
+        ebpf_map_definition_t definition;
+        uint16_t pin_path_length;
+        uint8_t pin_path[1];
+    } ebpf_serialized_map_information_t;
+
+    /**
+     * @brief Serialize array of ebpf_map_information_t onto output buffer.
+     *
+     * @param[in]  map_count Length of input array of ebpf_core_map_information_t structs.
+     * @param[in]  map_info Array of ebpf_map_information_t to serialize.
+     * @param[in]  output_buffer Caller specified output buffer to write serialized data into.
```

Comment:
Line 53 says this should be [out] not [in]

Reply:
Changed to [out].